script.video.randomtv/addon.py

```diff
@@ -0,0 +1,336 @@
+	else:
+		for includedShow in map(int, includedShows.split(", ")):
+			command = '{"jsonrpc": "2.0", "method": "VideoLibrary.GetEpisodes", "params": { "tvshowid": %d, "properties": ["showtitle", "file", "playcount", "lastplayed", "resume"] }, "id": 1}' % includedShow
+			allEpisodes = json.loads(xbmc.executeJSONRPC(command))
+			
+			if allEpisodes['result']['limits']['total'] > 0:
+				for episode in allEpisodes['result']['episodes']:
+					if addon.getSetting("IncludeUnwatched") == "true" or episode['playcount'] > 0:
+						log("Added Episode: " + str(episode['episodeid']) + " -- " + episode['showtitle'].encode('utf-8').strip().decode('utf-8') + " - " + episode['label'].encode('utf-8').strip().decode('utf-8'))
+						#log("Added Episode: " + episode['label'].encode('utf-8').strip())
+						myEpisodes.append({'episodeId': episode['episodeid'], 'episodeShow': episode['showtitle'].encode('utf-8').strip().decode('utf-8'), 'episodeName': episode['label'].encode('utf-8').strip().decode('utf-8'), 'episodeFile': episode['file'].encode('utf-8').strip().decode('utf-8'), 'playCount': episode['playcount'], 'lastPlayed': episode['lastplayed'], 'resume': episode['resume']})
+#		
+log("Total Episodes: " + str(len(myEpisodes)))
+
+
+# If no episodes, display notification and quit
+if len(myEpisodes) == 0:
+	log("--------- No episodes")
+	xbmcgui.Dialog().ok(name, addon.getLocalizedString(32008), addon.getLocalizedString(32009))
+	xbmc.executebuiltin('Addon.OpenSettings(%s)' % addonid)
+	backWindow.close()
+	log("Stopping")
+	log("-------------------------------------------------------------------------")
+	quit()
+else:
+	log("--------- Episodes Found")
+	# Get Auto Stop Check Time - Current Time + Auto Stop Check Timer
+	if addon.getSetting("AutoStop") == "true":
+		log("-- Auto Stop Enabled")
+		log("-- Auto Stop Timer: " + addon.getSetting("AutoStopTimer"))
+		log("-- Auto Stop Wait: " + addon.getSetting("AutoStopWait"))
+		AutoStopCheckTime = int(time.time()) + (int(addon.getSetting("AutoStopTimer")) * 60)
+		AutoStopWait = (int(addon.getSetting("AutoStopWait")) * 60)
+		AutoStopDialog = xbmcgui.DialogProgress()
+	#
+
+	# Initialize our Player
+	player = MyPlayer()
+	
+	# Create Playlist
+	myPlaylist = xbmc.PlayList(xbmc.PLAYLIST_VIDEO)
+
+	# Shuffle Episodes
+	random.shuffle(myEpisodes)
+	
+	# Build Playlist
+	buildPlaylist(myEpisodes)
+
+	# Start Player
+	player.play(item=myPlaylist)
+#
+
+
+while (not xbmc.Monitor().waitForAbort(1)):
+	if addon.getSetting("AutoStop") == "true":
+		if int(time.time()) >= AutoStopCheckTime:
+			log("-- Auto Stop Timer Reached")
+			AutoStopDialog.create(name, addon.getLocalizedString(32015))
+			while int(time.time()) < AutoStopCheckTime + AutoStopWait:
+				AutoStopDialog.update(int(int(time.time() - AutoStopCheckTime) * 100 / AutoStopWait), addon.getLocalizedString(32015), str(AutoStopWait - int(time.time() - AutoStopCheckTime)) + " " + addon.getLocalizedString(32016))
```

Comment:
`AutoStopDialog.update(int(int(time.time() - AutoStopCheckTime) * 100/ AutoStopWait)`
can be written as
`AutoStopDialog.update(int(((time.time() - AutoStopCheckTime) * 100)/ AutoStopWait))`
I went heavy with prentices so you can track the order... it can be trimed.

Reply:
Will have to test.  I have to imagine that 2017 me had a good reason for the double int() functions, but who knows, he did drink a lot..  :-)

Reply:
Making me re-look at this made me find that xbmc.DialogProgress.update no longer accepts line1, line2, and line3 arguments and now only accepts a single message argument.  :-)

Reply:
You are correct, I missed that during my review 🤗